protocols/kad/src/record.rs

```diff
@@ -122,6 +126,21 @@ impl Hash for ProviderRecord {
     }
 }
 
+impl PartialEq for ProviderRecord {
+    fn eq(&self, other: &Self) -> bool {
+        // In the [`Hash`] implementation of [`ProviderRecord`] all but the
+        // `key` and `provider` fields are ignored. The following property has
+        // to hold for all inputs: `k1 == k2 <=> hash(k1) == hash(k2)`. Thereby
+        // the [`PartialEq`] implementation has to ignore `key` and `provider`
+        // as well.
```

Comment:
While the change for equality seems fine, I think this comment is still misleading. This property does not need to hold in both directions and in general usually can't, and saying it must hold "modulo hash collisions" is rendering the property meaningless.

Reply:
Thanks for the input. I indeed followed a wrong reasoning here.

I have pushed 5f1c8bec93998ad2f5400150f2e6f18c0b00c087 which removes the comment, now only depending on the `Note` section added in https://github.com/libp2p/rust-libp2p/pull/1850/commits/1d28e21f94c84c552ef10f5011d393330bad381f.

Please insist in case I am still missing something.